builtin/providers/aws/resource_aws_s3_bucket_object.go

```diff
@@ -218,7 +230,22 @@ func resourceAwsS3BucketObjectRead(d *schema.ResourceData, meta interface{}) err
 	d.Set("content_language", resp.ContentLanguage)
 	d.Set("content_type", resp.ContentType)
 	d.Set("version_id", resp.VersionId)
-	d.Set("kms_key_id", resp.SSEKMSKeyId)
+	d.Set("server_side_encryption", resp.ServerSideEncryption)
+
+	// retrieve S3 KMS Default Master Key
+	kmsconn := meta.(*AWSClient).kmsconn
+	kmsresp, err := kmsconn.DescribeKey(&kms.DescribeKeyInput{
```

Comment:
I think there's no reason to call any KMS API at all if the S3 bucket/object isn't encrypted, i.e. if `SSEKMSKeyId` is `nil`.

It may look like an optimisation nitpick, but it's also about what IAM permissions we expect/require the user to have - in this case _everyone_ using `aws_s3_bucket_object` would _always_ need `kms:DescribeKey` even though they may not be using the encryption at all.